### PR TITLE
HP-1469/HW_summary_auto_generation

### PR DESCRIPTION
**Pull request by ValeriyShnurovoy**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new column in the server grid view to display differences in hardware summaries, enhancing the comparison capabilities.
	- Updated server model attributes to support new hardware summary functionalities.
	- Enhanced user interface in server views to show additional attributes based on user permissions.
	- Added a new library dependency to support visual rendering of differences in hardware summaries.

### CODE REVIEW

File: src/views/server/view.php [hunk 1]
```diff
@@ -482,7 +482,7 @@
         <?= Configuration::widget([
             'model' => $model,
             'configAttrs' => array_filter([
-                'summary',
+                'summary', 'summary_auto', 'comment',
```
[the remark this answers is not in the document]
Comment (tafid) [reply src/views/server/view.php:485]: согласен, исправь

**Review by tafid — CHANGES_REQUESTED** [review]

File: src/grid/ServerGridView.php [hunk 2]
```diff
@@ -444,6 +445,13 @@ public function columns()
                 'filterAttribute' => 'hwsummary_like',
                 'label' => Yii::t('hipanel:server', 'Hardware Summary'),
             ],
+            'hwsummary_auto' => [
+                'filter' => false,
+                'label' => Yii::t('hipanel:server', 'Auto Hardware Summary'),
+            ],
+            'hwsummary_diff' => [
+                'class' => SummaryDiffColumn::class,
```
Comment (SilverFire) [inline src/grid/ServerGridView.php:453]: I don't see a PR with this class

File: src/grid/SummaryDiffColumn.php [hunk 3]
```diff
@@ -0,0 +1,180 @@
+<?php
+
```
Comment (SilverFire) [inline src/grid/SummaryDiffColumn.php:2]: declare

File: src/grid/ServerGridView.php [hunk 4]
```diff
@@ -31,6 +31,7 @@
 use hipanel\modules\server\widgets\OSFormatter;
 use hipanel\modules\server\widgets\ResourceConsumptionTable;
 use hipanel\modules\server\widgets\State;
+use hipanel\modules\stock\grid\SummaryDiffColumn;
```
Comment (SilverFire) [inline src/grid/ServerGridView.php:34]: The class is in another namespace.
How does it work?

File: src/grid/ServerRepresentations.php [hunk 5]
```diff
@@ -49,6 +49,13 @@ protected function fillRepresentations(): void
                     'rack', 'client', 'dc', 'actions', 'server', 'hwsummary', 'hwcomment',
                 ],
             ] : null,
+            'summary' => $user->can('part.read') ? [
```
Comment (SilverFire) [inline src/grid/ServerRepresentations.php:52]:
```suggestion
            'summary' => $user->can('owner.staff') ? [
```

File: src/grid/SummaryDiffColumn.php [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        'lineNumbers' => false,
+        'spacesToNbsp' => true,
+        'resultForIdenticals' => true,
+        'wrapperClasses' => ['diff-wrapper'],
+    ];
+
+    public function init()
+    {
+        parent::init();
+        $this->grid->layout = $this->getCss() . $this->grid->layout;
```
Comment (SilverFire) [inline src/grid/SummaryDiffColumn.php:34]: Use `$this->grid->getView()->registerCss()` instead

**Review by SilverFire — CHANGES_REQUESTED** [review]

**Review by SilverFire — APPROVED** [review]

**Pull request merged**